api/src/opentrons/protocol_reader/config_analyzer.py

```diff
@@ -0,0 +1,88 @@
+"""Input file config analysis."""
+import ast
+from dataclasses import dataclass
+from typing import Union
+
+from opentrons.protocols.api_support.types import APIVersion
+from opentrons.protocols.parse import extract_metadata as extract_python_metadata
+from opentrons.protocols.models import JsonProtocol
+
+from .protocol_source import Metadata, PythonProtocolConfig, JsonProtocolConfig
+from .role_analyzer import RoleAnalyzedFile
+
+
+@dataclass(frozen=True)
+class ConfigAnalysis:
+    """Protocol config analyzed from main file."""
+
+    metadata: Metadata
+    config: Union[PythonProtocolConfig, JsonProtocolConfig]
+
+
+class ConfigAnalysisError(ValueError):
+    """Error raised if protocol config can not be determined."""
+
+
+class ConfigAnalyzer:
+    """Input file config analysis interface."""
+
+    @staticmethod
+    def analyze(main_file: RoleAnalyzedFile) -> ConfigAnalysis:
+        """Analyze the main file of a protocol to identify its config and metadata."""
+        if isinstance(main_file.data, JsonProtocol):
+            return ConfigAnalysis(
+                metadata=main_file.data.metadata.dict(exclude_none=True),
+                config=JsonProtocolConfig(schema_version=main_file.data.schemaVersion),
+            )
+
+        else:
+            return _analyze_python(main_file)
+
+
+# todo(mm, 2021-09-13): Deduplicate with opentrons.protocols.parse.
+def _analyze_python(main_file: RoleAnalyzedFile) -> ConfigAnalysis:  # noqa: C901
+    assert main_file.name.endswith(".py"), "Expected main_file to be Python"
+
+    try:
+        # todo(mm, 2021-09-13): Investigate whether it's really appropriate to leave
+        # the Python compilation flags at their defaults. For example, we probably
+        # don't truly want the protocol to inherit our own __future__ features.
+        module_ast = ast.parse(source=main_file.contents, filename=main_file.name)
+    except (SyntaxError, ValueError) as e:
+        # parse_python() raises SyntaxError for most errors,
```

Comment:
```suggestion
        # ast.parse() raises SyntaxError for most errors,
```

Renamed in the copy-paste, I think.